### PR TITLE
Add javadoc and inline comments to HashMap.java

### LINKED ISSUE #1394: Add javadoc and inline comments to HashMap.java
The file as it is now is pretty bare and needs to be well-commented. Adding these comments will make the code more readable and understandable to the user.

requirements:
Add a header to all public classes and methods
Add @return and @param where appropriate
Add a description of what the method/class does and why it is implemented
Follow oracle javadoc programming standards
Keep descriptions concise and to the point

optional:
free to add inline comments to clarify certain statements
free to add visual representation of the hash table with example cases

### PR DESCRIPTION
**Pull request by afadhitya**
I added a javadoc comment and made a little change to the modifier method

closes #1394

### CODE REVIEW

**Comment by rbshealy** [comment]
You may have to re-pull because upstream master HashMap.java just got updated

**Comment by afadhitya** [comment]
Hi @rbshealy please review this

Thanks

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+ * This class is an implementation of a hash table
+ * 
+ */
 public class HashMap {
     private int hsize;
     private LinkedList[] buckets;
 
+    /**
+     * Constructor initialize HashMap object and 
+     * create new Hashmap based on the size
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:13]: Add `@param hsize *description*`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 2]
```diff
@@ -13,39 +21,62 @@ public HashMap(int hsize) {
         this.hsize = hsize;
     }
 
-    public int hashing(int key) {
-        int hash = key % hsize;
-        if (hash < 0)
-            hash += hsize;
-        return hash;
-    }
-
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:22]: un-delete this

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 3]
```diff
@@ -73,7 +114,13 @@ public Node findKey(int key) {
     		}
     		
     	}
-    	
+			
+			/**
+			 * Find node based on the key and for certain node 
+			 * @param n a node which a member in linkedlist
+			 * @param key key of a node which will be find
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:121]: Use `"which will be found"`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 4]
```diff
@@ -73,7 +114,13 @@ public Node findKey(int key) {
...
+			 * @return
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:122]: Add `@return Node *Description*`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 5]
```diff
@@ -106,19 +164,31 @@ private void delete(Node n, int key) {
     			}
     		}
     	}
-    	
+			
+			/**
+			 * Call the display method  
+			 * @return
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:170]: Add `@return String *Description*`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 6]
```diff
@@ -127,20 +197,35 @@ public boolean isEmpty() {
     public static class Node {
     	private Node next;
     	private int key;
-    	
+			
+			/**
+			 * Constructor to create a new node and assign the initial key
+			 */
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:203]: Add `@param key *description*`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 7]
```diff
@@ -127,20 +197,35 @@ public boolean isEmpty() {
...
     	public Node(int key) {
     		next = null;
     		this.key = key;
     	}
-    	
+			
+			/**
+			 * Get the next node after the current node
+			 * @return next node
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:211]: Use `@return Node next node`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 8]
```diff
@@ -127,20 +197,35 @@ public boolean isEmpty() {
...
+			 */
     	public Node getNext() {
     		return next;
     	}
-    	
+			
+			/**
+			 * Get key of current node
+			 * @return key
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:219]: Use `@return int *description*`

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 9]
```diff
@@ -127,20 +197,35 @@ public boolean isEmpty() {
...
+			 */
     	public int getKey() {
     		return key;
     	}
-    	
+			
+			/**
+			 * Set next node of current node, make it chaining with current node 
+			 * @param next
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:227]: add a description to `@param next`

File: DataStructures/HashMap/Hashing/HashMap.java (hunk 2 above)
Comment (afadhitya) [reply DataStructures/HashMap/Hashing/HashMap.java:22]: Actually for this thing I've change it to private

**Comment by afadhitya** [comment]
Hi @rbshealy I have an updated

**Comment by rbshealy** [comment]
Looks great! The only thing left is to un-delete the hashing method

**Comment by afadhitya** [comment]
Hi @rbshealy i've done un-delete hashing method

**Comment by rbshealy** [comment]
great work! This addresses the issue pretty well 👍🏻

**Comment by afadhitya** [comment]
Hi @StepfenShawn could you please review then merge this if necessary?
Thanks in advance

File: DataStructures/HashMap/Hashing/HashMap.java [hunk 10]
```diff
@@ -127,20 +201,36 @@ public boolean isEmpty() {
...
     	public Node(int key) {
     		next = null;
     		this.key = key;
     	}
-    	
+			
+			/**
+			 * Get the next node after the current node
+			 * @return Node next node
+			 */
```
Comment (rbshealy) [inline DataStructures/HashMap/Hashing/HashMap.java:217]: Fix indentation example

Comment (rbshealy) [reply DataStructures/HashMap/Hashing/HashMap.java:217]: Before this gets merged, I just re-checked it and found some indentation errors, can you fix these?

Comment (rbshealy) [reply DataStructures/HashMap/Hashing/HashMap.java:217]: @afadhitya

**Pull request closed without merging**